pkg/tsdb/cloud-monitoring/annotation_query.go

```diff
@@ -39,10 +42,14 @@ func (s *Service) executeAnnotationQuery(ctx context.Context, req *backend.Query
 	firstQuery := req.Queries[0]
 	err = json.Unmarshal(firstQuery.JSON, &tslq)
 	if err != nil {
+		logger.Error("error unmarshaling query", "error", err, "statusSource", backend.ErrorSourceDownstream)
+		errorsource.AddErrorToResponse(firstQuery.RefID, resp, err)
 		return resp, nil
 	}
-	err = parseToAnnotations(req.Queries[0].RefID, queryRes, dr.(cloudMonitoringResponse), tslq.TimeSeriesList.Title, tslq.TimeSeriesList.Text)
-	resp.Responses[firstQuery.RefID] = *queryRes
+
+	// parseToAnnotations never actually returns an error
+	err = parseToAnnotations(req.Queries[0].RefID, dr, queryRes.(cloudMonitoringResponse), tslq.TimeSeriesList.Title, tslq.TimeSeriesList.Text)
+	resp.Responses[firstQuery.RefID] = *dr
```

Comment:
should this happen after the error check?

Reply:
Yeah this is as it was before I just renamed a variable. It's not even possible for `parseToAnnotations` to return an error so I think we should remove the error check in the future.